apps/aecore/lib/aecore/peers/worker.ex

```diff
@@ -113,44 +144,57 @@ defmodule Aecore.Peers.Worker do
   is updated if the one in the latest GET /info request is different.
   """
   def handle_call(:check_peers, _from, %{peers: peers} = state) do
-    filtered_peers = :maps.filter(fn(peer, _) ->
-        case Client.get_info(peer) do
-          {:ok, info} -> info.genesis_block_hash == genesis_block_header_hash()
-          _ -> false
+    filtered_peers = :maps.filter(fn(_, %{uri: uri}) ->
+        case Client.get_info(uri) do
+          {:ok, info} ->
+            info.genesis_block_hash == genesis_block_header_hash()
+          _ ->
+            false
         end
       end, peers)
-
     updated_peers =
-      for {peer, current_block_hash} <- filtered_peers, into: %{} do
-        {_, info} = Client.get_info(peer)
-        if info.current_block_hash != current_block_hash do
-          {peer, info.current_block_hash}
+      for {nonce, %{uri: uri, latest_block: latest_block}} <- filtered_peers, into: %{} do
+        {_, info} = Client.get_info(uri)
+        if(info.current_block_hash != latest_block) do
```

Comment:
Please remove unnecessary parenthesis.

Reply:
which parenthesis are unnecessary?

Reply:
around statement after if. It should be:
`if info.current_block_hash != latest_block do`